core/src/main/java/brooklyn/entity/basic/AbstractEntity.java

```diff
@@ -962,7 +962,9 @@ public void removeAttribute(AttributeSensor<?> attribute) {
     // -------- CONFIGURATION --------------
 
     @Override
-    public ConfigurationSupportInternal config() {
+    // the concrete type rather than an interface is returned because Groovy subclasses
+    // complain (incorrectly) if we return ConfigurationSupportInternal 
+    public BasicConfigurationSupport config() {
```

Comment:
Does the same apply to `AbstractEntityAdjunct.config()`, which also returns `ConfigurationSupportInternal`?
And same for `AbstractLocation.config()`?